logdb/logdb_bench_test.go

```diff
@@ -68,90 +46,36 @@ func BenchmarkFakeDB_NewestBlockID(t *testing.B) {
 	receipts := tx.Receipts{newReceipt()}
 
 	w := db.NewWriter()
-	if err := w.Write(b, receipts); err != nil {
-		t.Fatal(err)
-	}
-	if err := w.Commit(); err != nil {
-		t.Fatal(err)
-	}
+	require.NoError(t, w.Write(b, receipts))
+	require.NoError(t, w.Commit())
 
 	tests := []struct {
 		name    string
 		prepare func() (thor.Bytes32, error)
 	}{
 		{
 			"newest block id",
-			func() (thor.Bytes32, error) {
-				return b.Header().ID(), nil
-			},
-		},
-		{
-			"add both event and transfer, best should change",
 			func() (thor.Bytes32, error) {
 				b = new(block.Builder).
 					ParentID(b.Header().ID()).
-					Transaction(newTx()).
 					Build()
 				receipts := tx.Receipts{newReceipt()}
 
-				w := db.NewWriter()
-				if err := w.Write(b, receipts); err != nil {
-					return thor.Bytes32{}, nil
-				}
-				if err := w.Commit(); err != nil {
-					return thor.Bytes32{}, nil
-				}
-				return b.Header().ID(), nil
-			},
-		},
-		{
-			"add event only, best should change",
-			func() (thor.Bytes32, error) {
-				b = new(block.Builder).
-					ParentID(b.Header().ID()).
-					Transaction(newTx()).
-					Build()
-				receipts := tx.Receipts{newEventOnlyReceipt()}
-
-				w := db.NewWriter()
-				if err := w.Write(b, receipts); err != nil {
-					return thor.Bytes32{}, nil
-				}
-				if err := w.Commit(); err != nil {
-					return thor.Bytes32{}, nil
-				}
-				return b.Header().ID(), nil
-			},
-		},
-		{
-			"add transfer only, best should change",
-			func() (thor.Bytes32, error) {
-				b = new(block.Builder).
-					ParentID(b.Header().ID()).
-					Transaction(newTx()).
-					Build()
-				receipts := tx.Receipts{newTransferOnlyReceipt()}
+				require.NoError(t, w.Write(b, receipts))
+				require.NoError(t, w.Commit())
 
-				w := db.NewWriter()
-				if err := w.Write(b, receipts); err != nil {
-					return thor.Bytes32{}, nil
-				}
-				if err := w.Commit(); err != nil {
-					return thor.Bytes32{}, nil
-				}
 				return b.Header().ID(), nil
 			},
 		},
 	}
 
-	t.ResetTimer()
 	for _, tt := range tests {
 		t.Run(tt.name, func(b *testing.B) {
+			t.ResetTimer()
```

Comment:
Reset timer here will result the benchmark only record the last run.

Reply:
Fixed 👍